packages/atomic/src/decorators/with-tailwind-styles.ts

```diff
@@ -3,13 +3,34 @@ import theme from '@/src/utils/coveo.tw.css';
 import styles from '@/src/utils/tailwind.global.tw.css';
 import utilities from '@/src/utils/tailwind-utilities/utilities.tw.css';
 
+let tailwindPropertiesInjected = false;
+
+function injectTailwindProperties() {
+  if (typeof document === 'undefined') return;
+  if (tailwindPropertiesInjected) return;
+  const propCss = styles as unknown as string[] | string;
+  const fullCss = Array.isArray(propCss) ? propCss.join('') : propCss;
+
+  // Extract only the variable definitions inside @layer properties
+  const match = fullCss.match(
+    /@layer\s+properties\s*{[^{}]*@supports[^{}]*{([^}]*)}|@layer\s+properties\s*{([^}]*)}/
```

Comment:
This will match literally `@layer properties { /* ... */ }`, not say `@layer base { /* ... */ }`

Is this really what we want?